juriscraper/opinions/united_states/federal_bankruptcy/bap1.py

```diff
@@ -0,0 +1,100 @@
+"""
+Scraper for the United States Bankruptcy Appellate Panel for the First Circuit
+CourtID: bap1
+Court Short Name: 1st Cir. BAP
+"""
+from datetime import datetime, timedelta
+from typing import Dict
+
+from lxml.html import HtmlElement
+
+from juriscraper.AbstractSite import logger
+from juriscraper.OpinionSiteLinear import OpinionSiteLinear
+
+
+class Site(OpinionSiteLinear):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.url = "https://www.bap1.uscourts.gov/bapopn"
```

Comment:
drop all the parameters stuff from the normal scraper.  and write a simple method that encodes the dates into the URL for the non-back scraper tool

Reply:
Since the base code is designed to pass that "parameters" dictionary to the requests call, I think it is better to use it that way. I have refactored it into a function, though. In general, doing manually what the library would do anyway seems to me like a bad practice

Reply:
Thanks @grossir I appreciate your thoughts.